test/core/crawlers/basic_crawler.test.ts

```diff
@@ -267,6 +268,49 @@ describe.each(SingleStorageCase)('BasicCrawler - %s', (Emulator) => {
         expect(await requestList.isEmpty()).toBe(true);
     });
 
+    test('should use errorHandler', async () => {
+        const sources = [{ url: 'http://example.com/', label: 'start' }];
+
+        let errorHandlerCalls = 0;
+        let failedRequestHandlerCalls = 0;
+
+        const failed: Dictionary<{ request: Request; error: Error }> = {};
+        const requestList = new RequestList({ sources });
+
+        const requestHandler: RequestHandler = async ({ request }) => {
+            expect(request.label).toBe(errorHandlerCalls === 0 ? 'start' : `error_${errorHandlerCalls}`);
+            throw new Error(`This is an error ${errorHandlerCalls}`);
+        };
+
+        const errorHandler: ErrorHandler = async ({ request }, error) => {
+            expect(error.message).toBe(`This is an error ${errorHandlerCalls}`);
+            errorHandlerCalls++;
+            request.label = `error_${errorHandlerCalls}`;
+        };
+
+        const failedRequestHandler: FailedRequestHandler = async ({ request, error }) => {
+            failed[request.url] = { request, error };
+            failedRequestHandlerCalls++;
+        };
+
+        const basicCrawler = new BasicCrawler({
+            requestList,
+            requestHandler,
+            errorHandler,
+            failedRequestHandler,
+        });
+
+        await requestList.initialize();
+        await basicCrawler.run();
```

Comment:
random thought: i saw this pattern in tests quite often, maybe we could run `requestList.initialize()` automatically in `crawler.run()`, wdyt @mnmkng?

Reply:
Actually, `openRequestList` does not exist anymore, right? 😄

Reply:
I actually put it there just because it's kinda widely used in the tests, so I thought maybe there's some deeper meaning into this inside of the tests :) Will update it (at least here).

Reply:
#157

Reply:
@B4nan I was just cleaning up some tests (basically doing the same) =D Need to go for like an hour... Will finish with tests right after and it should be it :)